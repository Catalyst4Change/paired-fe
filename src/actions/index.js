export const setLoading = bool => ({
  type: 'SET_LOADING',
  bool
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});