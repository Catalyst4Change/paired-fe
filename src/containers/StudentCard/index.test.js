import React from 'react';
import { shallow } from 'enzyme';
import { StudentCard, mapDispatchToProps } from './index';
import {
  mockStudentForStudentCard,
  mockStudForStudentCardNoAvail,
  mockStudForStudentCardAllAvail
} from '../../mockData';
import { setPairingId } from '../../actions';

describe('StudentCard', () => {
  describe('StudentCard component', () => {
    let wrapper;
    const mockSetPairingId = jest.fn();
    const mockPropsAllTimesAvail = mockStudForStudentCardAllAvail;
    const mockPropsNoTimes = mockStudForStudentCardNoAvail;
    const mockHistory = { push: jest.fn() };
    beforeEach(() => {
      wrapper = shallow(
        <StudentCard
          student={mockPropsAllTimesAvail}
          setPairingId={mockSetPairingId}
          history={mockHistory}
        />
      );
    });

    it('should match the snapshot when the student has all times available', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the student has some times available', () => {
      wrapper = shallow(
        <StudentCard
          student={mockStudentForStudentCard}
          setPairingId={mockSetPairingId}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the student has no available times', () => {
      wrapper = shallow(
        <StudentCard
          student={mockPropsNoTimes}
          setPairingId={mockSetPairingId}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleClick when the Book It button for morning is clicked', () => {
      const expectedId = mockStudForStudentCardAllAvail.morning;
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handleClick');
      wrapper.find('#morning').simulate('click');
      expect(instance.handleClick).toHaveBeenCalledWith(expectedId); 
    });

    it('should call handleClick when the Book It button for lunch is clicked', () => {
      const expectedId = mockStudForStudentCardAllAvail.lunch;
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handleClick');
      wrapper.find('#lunch').simulate('click');
      expect(instance.handleClick).toHaveBeenCalledWith(expectedId);
    });

    it('should call handleClick when the Book It button for afternoon is clicked', () => {
      const expectedId = mockStudForStudentCardAllAvail.afternoon;
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handleClick');
      wrapper.find('#afternoon').simulate('click');
      expect(instance.handleClick).toHaveBeenCalledWith(expectedId);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with setPairingId when setPairingId is called', () => {
      const mockDispatch = jest.fn();
      const mockId = '4wkljf';
      const actionToDispatch = setPairingId(mockId);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setPairingId(mockId);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
