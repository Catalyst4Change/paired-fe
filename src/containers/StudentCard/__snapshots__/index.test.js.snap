// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCard StudentCard component should match the snapshot when the student has all times available 1`] = `
<section
  className="StudentCard"
>
  <h3
    className="StudentCard--header"
  >
    Hillary
     (
    she/her
    )
  </h3>
  <div
    className="StudentCard--profile"
  >
    <img
      alt="Hillary"
      className="StudentCard--image"
      src="http://www.image.com/hill.jpeg"
    />
    <div
      className="StudentCard--skills-div"
    >
      <p
        className="StudentCard--skills-p"
      >
        Skills
      </p>
      <ul
        className="StudentCard--ul"
      >
        <li
          className="StudentCard-li"
          key="0"
        >
          react
        </li>
        <li
          className="StudentCard-li"
          key="1"
        >
          redux
        </li>
      </ul>
    </div>
  </div>
  <div
    className="StudentCard--div--rows"
  >
    <div>
      <p
        className="StudentCard-time-text"
      >
        8:00 - 8:50 a.m.
      </p>
      <button
        className="StudentCard--btn"
        disabled={false}
        id="morning"
        onClick={[Function]}
      >
        Book Pairing
      </button>
    </div>
    <div>
      <p
        className="StudentCard-time-text"
      >
        12:00 - 12:50 p.m.
      </p>
      <button
        className="StudentCard--btn"
        disabled={false}
        id="lunch"
        onClick={[Function]}
      >
        Book Pairing
      </button>
    </div>
    <div>
      <p
        className="StudentCard-time-text"
      >
        4:10 - 5:00 p.m.
      </p>
      <button
        className="StudentCard--btn"
        disabled={false}
        id="afternoon"
        onClick={[Function]}
      >
        Book Pairing
      </button>
    </div>
  </div>
</section>
`;

exports[`StudentCard StudentCard component should match the snapshot when the student has no available times 1`] = `
<section
  className="StudentCard"
>
  <h3
    className="StudentCard--header"
  >
    Hillary
     (
    she/her
    )
  </h3>
  <div
    className="StudentCard--profile"
  >
    <img
      alt="Hillary"
      className="StudentCard--image"
      src="http://www.image.com/hill.jpeg"
    />
    <div
      className="StudentCard--skills-div"
    >
      <p
        className="StudentCard--skills-p"
      >
        Skills
      </p>
      <ul
        className="StudentCard--ul"
      >
        <li
          className="StudentCard-li"
          key="0"
        >
          react
        </li>
        <li
          className="StudentCard-li"
          key="1"
        >
          redux
        </li>
      </ul>
    </div>
  </div>
  <div
    className="StudentCard--div--rows"
  >
    <div>
      <p
        className="StudentCard-time-text"
      >
        8:00 - 8:50 a.m.
      </p>
      <p>
        NOT AVAILABLE
      </p>
    </div>
    <div>
      <p
        className="StudentCard-time-text"
      >
        12:00 - 12:50 p.m.
      </p>
      <p>
        NOT AVAILABLE
      </p>
    </div>
    <div>
      <p
        className="StudentCard-time-text"
      >
        4:10 - 5:00 p.m.
      </p>
      <p>
        NOT AVAILABLE
      </p>
    </div>
  </div>
</section>
`;

exports[`StudentCard StudentCard component should match the snapshot when the student has some times available 1`] = `
<section
  className="StudentCard"
>
  <h3
    className="StudentCard--header"
  >
    Hillary
     (
    she/her
    )
  </h3>
  <div
    className="StudentCard--profile"
  >
    <img
      alt="Hillary"
      className="StudentCard--image"
      src="http://www.image.com/hill.jpeg"
    />
    <div
      className="StudentCard--skills-div"
    >
      <p
        className="StudentCard--skills-p"
      >
        Skills
      </p>
      <ul
        className="StudentCard--ul"
      >
        <li
          className="StudentCard-li"
          key="0"
        >
          react
        </li>
        <li
          className="StudentCard-li"
          key="1"
        >
          redux
        </li>
      </ul>
    </div>
  </div>
  <div
    className="StudentCard--div--rows"
  >
    <div>
      <p
        className="StudentCard-time-text"
      >
        8:00 - 8:50 a.m.
      </p>
      <p>
        NOT AVAILABLE
      </p>
    </div>
    <div>
      <p
        className="StudentCard-time-text"
      >
        12:00 - 12:50 p.m.
      </p>
      <button
        className="StudentCard--btn"
        disabled={false}
        id="lunch"
        onClick={[Function]}
      >
        Book Pairing
      </button>
    </div>
    <div>
      <p
        className="StudentCard-time-text"
      >
        4:10 - 5:00 p.m.
      </p>
      <p>
        NOT AVAILABLE
      </p>
    </div>
  </div>
</section>
`;
